### PR TITLE
OAuth 2.0 PKCE challenge (RFC7636)

### LINKED ISSUE #139: Support PKCE for OAuth 2.0 public clients
> PKCE ([RFC 7636](http://tools.ietf.org/html/rfc7636)) is an extension to the [Authorization Code flow](https://oauth.net/2/grant-types/authorization-code/) to prevent CSRF and authorization code injection attacks.

### Reference
- https://datatracker.ietf.org/doc/html/rfc7636
- https://oauth.net/2/pkce/

### PR DESCRIPTION
**Pull request by byewokko**
closes #139

PKCE is always enabled in Seacat Auth; it is not necessary to configure the app.

Both `plain` and `S256` code challenge methods are supported.

To be able to use PKCE with your client:
- Register your client and set its `code_challenge_methods` attribute to `["S256"]` (or `["plain"]` if SHA256 is not implemented on the client server)
- A client is allowed to register only one method at once, i.e. it is not possible to set `{"code_challenge_methods": ["S256", "plain"]}`

The PKCE code challenge works as described in [RFC7636](https://datatracker.ietf.org/doc/html/rfc7636#section-4)

### CODE REVIEW

File: seacatauth/client/service.py [hunk 1]
```diff
@@ -15,9 +15,9 @@
 
 #
 
-# TODO: Implement support for remaining metadata
-# TODO: Implement support for other response and grant types and stuff
 # https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata
+# TODO: This is not Client responsibility!
```
Comment (ateska) [inline seacatauth/client/service.py:19]: What is "This" here?

**Review by ateska — CHANGES_REQUESTED** [review]
Please also include PKCE in the README.md as a supported feature - including link to the PKCE specifications.

File: seacatauth/client/service.py (hunk 1 above)
Comment (byewokko) [reply seacatauth/client/service.py:19]: this line is probably not necessary, i will delete it. the following line explains it clearly.

**Review by ateska — APPROVED** [review]

File: seacatauth/client/service.py (hunk 1 above)
Comment (ateska) [reply seacatauth/client/service.py:19]: ok

**Comment by ateska** [comment]
https://github.com/TeskaLabs/seacat-auth/blob/main/README.md updated also.

**Pull request merged**